frontend/components/YourWalletPage/index.tsx

```diff
@@ -48,47 +65,76 @@ const Address = () => {
 };
 
 const OlasBalance = () => {
-  const { masterSafeBalance: safeBalance, totalOlasStakedBalance } =
-    useBalanceContext();
+  const { masterSafes } = useMasterWalletContext();
+  const { walletBalances, totalStakedOlasBalance } = useBalanceContext();
+
+  const masterSafeOlasBalance: number = walletBalances
+    ?.filter((balance) => balance.symbol === TokenSymbol.OLAS)
+    .reduce((acc, balance) => acc + balance.balance, 0);
+
   const olasBalances = useMemo(() => {
     return [
       {
         title: 'Available',
-        value: balanceFormat(safeBalance?.OLAS ?? 0, 2),
+        value: balanceFormat(masterSafeOlasBalance ?? 0, 2),
       },
       {
         title: 'Staked',
-        value: balanceFormat(totalOlasStakedBalance ?? 0, 2),
+        value: balanceFormat(totalStakedOlasBalance ?? 0, 2),
       },
     ];
-  }, [safeBalance?.OLAS, totalOlasStakedBalance]);
+  }, [masterSafeOlasBalance, totalStakedOlasBalance]);
+
+  if (!masterSafes) return <Skeleton />;
+  if (isEmpty(masterSafes)) return null;
 
   return (
     <Flex vertical gap={8}>
-      <Text strong>OLAS</Text>
+      <Text strong>{TokenSymbol.OLAS}</Text>
       <InfoBreakdownList
         list={olasBalances.map((item) => ({
           left: item.title,
           leftClassName: 'text-light',
-          right: `${item.value} OLAS`,
+          right: `${item.value} ${TokenSymbol.OLAS}`,
         }))}
         parentStyle={infoBreakdownParentStyle}
       />
     </Flex>
   );
 };
 
-const XdaiBalance = () => {
-  const { masterSafeBalance: safeBalance } = useBalanceContext();
+const MasterSafeNativeBalance = () => {
+  const { masterSafes, masterEoa } = useMasterWalletContext();
+  const { walletBalances } = useBalanceContext();
+
+  const masterSafeNativeBalance: Optional<number> = useMemo(() => {
+    if (isNil(masterSafes)) return;
+    if (isNil(walletBalances)) return;
+
+    if (isEmpty(masterSafes)) return 0;
+    if (isEmpty(walletBalances)) return 0;
+
+    const masterSafe = masterSafes[0]; // TODO: handle multiple safes in future
+
+    return walletBalances
```

Comment:
maybe can use `useMasterBalances` instead? and then maybe only check isLoaded from useBalanceContext ?